src/test/java/com/currencycloud/client/UtilsTest.java

```diff
@@ -15,9 +23,35 @@ public void returnsRootCauseForChain() throws Exception {
                 equalTo((Throwable) root)
         );
     }
+
     @Test
     public void returnsRootCauseForCauseless() throws Exception {
         RuntimeException e = new RuntimeException();
-        assertThat(Utils.getRootCause(e), equalTo((Throwable)e));
+        assertThat(Utils.getRootCause(e), equalTo((Throwable) e));
+    }
+
+    @Test
+    public void joinInverse_validListAndSeparator_listInverted() {
+        String result = Utils.joinInverse(ANIMALS, "|");
+        assertThat(result, is("pig|cow|elephant"));
+    }
+
+    @Test
+    public void joinInverse_singleItemInList_itemReturned() {
+        String item = "kangaroo";
+        String result = Utils.joinInverse(Collections.singletonList(item), "|");
+        assertThat(result, is(item));
+    }
+
+    @Test
+    public void joinInverse_nullList_nullReturned() {
+        String result = Utils.joinInverse(null, "|");
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void joinInverse_nullSeparator_nullReturned() {
```

Comment:
this is a change in behaviour. previously, `null` being fed in would mean a string of `null` would be used as a separator. perhaps a better behaviour would be a `null` separator lead to the strings being reversed and concatenated